test/clj/cider/nrepl/middleware/slurp_test.clj

```diff
@@ -26,3 +26,10 @@
     (t/is (= ["application/octet-stream" {}] (:content-type resp)))
     (t/is (str/starts-with? (:body resp) "#binary[location="))
     (t/is (str/ends-with? (:body resp) ",size=681]"))))
+
+(t/deftest test-directory
```

Comment:
Don't we also need for whatever was causing the NPE?

Reply:
The problem was some URLs return nil when doing `.getContentType`. I couldn't really figure out why the URLs did not return a content type. 
I thought it was because of redirection but not all redirected URLs return nil. Tried various HTTP return codes. No idea why this happens.